### ISSUE TITLE
work on citations connections figure

### ISSUE THREAD

**Issue by lizzieinvancouver** [lizzieinvancouver/mastjc#2]
JHRL showed me how to do this today from ISI (she did it in this [paper: Accidental experiments: ecological and evolutionary insights and opportunities derived from global change](https://nsojournals.onlinelibrary.wiley.com/doi/pdf/10.1111/j.1600-0706.2013.00698.x) You basically:

1. ISI Web of Science (can only access through library)
2. You end up on 'Documents' but need to click on 'Cited references'
3. From there, create a search for each paper or sets of papers you are interested in. For example: Add `Janzen, DH` for author and 'seeds' into the next field (pick a relevant field from drop down like 'title') and then you can find in the list 'Seed predation by animals' (cited 1509 times). 
4. CLICK the box next to all articles you want together. 
5. CLICK 'see results'
6. Now repeat previous three steps for the other dataset you're interested (e.g., find the original papers behind Janzen and Connell) 
7. Now go to'Search History' (often on the left, you may need to click on 'menu' to see it)
8. Here you can combine searches. For example, combine using 'AND' to see papers that cite two different literatures (e.g., masting AND JC). [Update: You may need to go to Advanced Search from this page to do this.]
9. From there you can export results and do other stuff ... we need to figure the rest out together.

**Comment by lizzieinvancouver** [comment]
Helpful hint: Do this when logged into an ISI account (Lizzie has one) so we can save all searches etc.

**Comment by lizzieinvancouver** [comment]
Jonathan wants: 

1. search for: all the papers about "density-dependen*" that cite Janzen 1971 (Annual Review) or Janzen 1970 (Herbivores and the number of tree species in tropical forests) -- get total number of papers 
2. search for: all the papers about "predator satiat*" that cite Janzen 1971 or Janzen 1970 (Herbivores and the number of tree species in tropical forests) -- get total number of papers

**Comment by lizzieinvancouver** [comment]
I would also like to know:
3. search for: all the papers about "negative density-dependen*" that cite Janzen 1971 (Annual Review) or Janzen 1970 (Herbivores and the number of tree species in tropical forests) -- get total number of papers (we got around 382 working on this together me and @ngoj1 )
4. search for: all the papers about "masting" that cite Janzen 1971 or Janzen 1970 (Herbivores and the number of tree species in tropical forests) -- get total number of papers
5.  search for: all the papers about "negative density-dependen*" that cite Janzen 1970 (Herbivores and the number of tree species in tropical forests) -- get total number of papers
6. search for: all the papers about "predator satiat*" that cite Janzen 1971 -- get total number of papers
7. search for: all the papers about "masting*" that cite Janzen 1971 -- get total number of papers

@ngoj1 I was hoping you could help with this, so let's discuss today.

**Comment by lizzieinvancouver** [comment]
Papers are:
*  Janzen 1970 (Herbivores and the number of tree species in tropical forests) in Am Nat
* Jansen 1971 Seed predation by animals in Annual Rev of Eco and Sys

You could also look into how to map citation networks in R (maybe by decade?) to get a sense of data we eventually would want.

**Comment by ngoj1** [comment]
After trying many times to get my advanced search to cooperate (whenever I want to combine sets through and/or it just freezes up and says invalid search), I decided to just manually add the numbers together.

Jonathan's search request:
 - "density-dependen*" citing Janzen 1970 or 1971: **779 papers**
 - "predator satiat*" citing Janzen 1970 or 1971: **193 papers**
 
Lizzie's search request:
 - "negative density-dependen*" citing Janzen 1970 or 1971: **270 papers** (um...this is not 382...I'll explain my method below)
 - "masting" citing Janzen 1970 or 1971: **241 papers**
 - "negative density-dependen*" citing Janzen 1970 ONLY: **257 papers**
 - "predator satiat*" citing Janzen 1971 ONLY: "predator satiat*": **148 papers**
 - "masting" citing Janzen 1971 ONLY: **196 papers**

What I did to get these numbers is I went to the Web of Science database, looked for the Janzen papers by inputting his name and the year of publication in the Cited References search engine (which gave me 3,347 papers for Janzen 1970 and 1,490 for Janzen 1971). Clicking on either of these would direct me to the search results, and I would then use the search engine on the left to look for the key terms (e.g. `"predator satiat*"`) which gave me the number of papers concerned with this term within the list of ALL papers that had cited the Janzen paper I was currently focusing on. 

I wonder if there are some papers that are only accessible to authorized faculty? I made a new account under the temporalecologylab@gmail.com account, and maybe it's not recognized/verified? Alternatively, I might have forgotten one crucial step when you were demonstrating how this Cited References search engine works and I'm just doing it wrong.

**Comment by lizzieinvancouver** [comment]
@ngoj1 Thanks for working on this and I am so sorry this did not work for you! I think there likely was a step missing so I did this myself today and wrote out all the steps. Did you perhaps missing clicking on `Advanced Search`? 

I got:
- 3350 for Herbivores and the number of tree species in tropical forests (original, plus 3 additional mis-citations). 
- 1493 for Seed Predation by animals (origina, plus 2 mis-citations)

Here's what I did:
1. ISI Web of Science (can only access through library)
2. You end up on 'Documents' but need to click on 'Cited references' to start
3. From there, create a search for each paper or sets of papers you are interested in. For example: Add `Janzen, DH` for author and 'seeds' into the next field (pick a relevant field from drop down like 'title') and then you can find in the list 'Seed predation by animals' (cited 1509 times). 
4. CLICK on each box next to all articles you want together. 
5. CLICK 'see results'
6. Repeat for other Janzen article
7. Now go back to search and click on `Documents` header and search for terms (e.g., predator satiat* using TOPIC)
8. Now, on left click on 'View your Search History' ... here I see all my searches, including when I clicked 'see results' for the exact Janzen citations. 
9. I now click on Advanced Search which is next to the filter by date range -- To combine searches, go to `Advanced search` it says. Here I see only the results searches, which are exactly what I want to combine. 
10. Click the open box next to what I want to combine (two at most, so combine the two Janzen ones first for Jonathan's request using OR). Once the boxes are clicked, you need to click on `combine sets` to pick if you want to ADD or OR them. 

Here's what I get:

1. (Janzen 1971 OR Janzen 1970) AND predator satiat* = 202 records
2. (Janzen 1971 OR Janzen 1970) AND density-dependen* = 743 records
3. (Janzen 1971 OR Janzen 1970) AND negative density-dependen* = 366 (not quite the 382 I got before, should double check my Janzen collections!)
4. (Janzen 1971 OR Janzen 1970) AND masting = 341 
5. (Janzen 1970 Herbivores and the number of tree species in tropical forests) AND density-dependen* = 714
6. (Janzen 1971) AND "predator satiat*"  = 182
7. (Janzen 1971) AND masting = 289

**Comment by lizzieinvancouver** [comment]
@ngoj1 I think I can download the info we need from ISI if you're still struggling but could use your help to figure out what to download. We'd like to make a figure looking at citation network connectance within and between (for now, we may tweak later but getting the basic method down would be good):

1. (Janzen 1971 OR Janzen 1970) AND density-dependen* = 743 records
2. (Janzen 1971 OR Janzen 1970) AND predator satiat* = 202 records

So basically how much do papers within group 1 cite themselves versus papers in group 2 and ditto for group 2. 

Can you see if you can figure out how to do this? It may be that we need to ask a librarian and/or look at papers working on citations networks to figure out what they do....

**Comment by ngoj1** [comment]
> Now go back to search and click on Documents header and search for terms (e.g., predator satiat* using TOPIC)
Now, on left click on 'View your Search History' ... here I see all my searches, including when I clicked 'see results' for the exact Janzen citations.
I now click on Advanced Search which is next to the filter by date range -- To combine searches, go to Advanced search it says. Here I see only the results searches, which are exactly what I want to combine.

Oh, I see where I went wrong. I was using the search results for a given Janzen paper and looking through those for "masting" or "density dependen*" etc. instead of opening a new search for those key terms, which is also why my "combine sets" didn't work. I tried this again and somehow got 383 for ((Janzen 1971 OR Janzen 1970) AND "negative density dependen*") which is more than your 366 and also more than the original 382, so I wonder what's going on there...in any case, I got pretty much the same numbers as you did. 

> So basically how much do papers within group 1 cite themselves versus papers in group 2 and ditto for group 2.

I was looking through the search and advanced search options to see if there's a way to compare citations within the combined set, but it doesn't look like there's a straightforward way...right now the Library assistance chat is offline, but I can ask them tomorrow to see if they know how to do this and maybe arrange a meeting with them in person so I can better explain what we want. I'll keep you updated as soon as I get new information!

**Comment by lizzieinvancouver** [comment]
@ngoj1 Great! Your 383 sounds good (better than my 366, I am hopeful the difference is how many of the messed up citations are selected). 

I think we probably need to do something like download all the citations WITHIN the paper sets and compare those, but am not sure. However, some googling turned up the following possibly useful websites, you might be able to find better, but these seem a start:

- [This video](https://www.youtube.com/watch?v=UU2M9xdci84) goes over some open source cites and how to do this in Python -- those two things we may not want, but it seems to show the TYPE of data we may need and HOW to format it etc., so could be a good reference.
- [This question and answer seems helpful](https://datascience.stackexchange.com/questions/26674/building-a-citation-network-to-analyze-in-r)

I think basically we need to figure out the format of the data we need *and* how to plot networks in R, so it could be better to search each of these separately. Let me know how it goes. It would be great to have a basic graph in two weeks, but it might be better as a collaborative project in which case I could ask someone else to help you with it. Let me know how it goes over the next few days and we can decide.

**Comment by ngoj1** [comment]
I've downloaded a list of the 743 and 202 papers for each respective group, and now just trying to figure out how to download a list of all the papers they cite (or alternatively, the papers which cite them). Web of Science lets me save 50 papers at a time to my marked list which then lets me view the total list of **references** (which, for 50 papers, is around 1900). 

![image](https://github.com/lizzieinvancouver/mastjc/assets/87724921/7bc08872-e049-4252-b1b1-dbca1f545d72)
![image](https://github.com/lizzieinvancouver/mastjc/assets/87724921/31185ee3-7870-43ec-b7b7-b1d60e75a8ff)
![image](https://github.com/lizzieinvancouver/mastjc/assets/87724921/700f49b6-02ff-4d3d-b1a9-9e62e4444989)

I can then just download this list as before and make my "citation" list as that StackExchange Q&A indicate. 

I asked a librarian on the UBC library's online chat but unfortunately they weren't familiar with this, so if I can't figure out a better method anytime soon I will probably visit them in person since it's difficult to explain our question through text only.

**Comment by lizzieinvancouver** [comment]
> I can then just download this list as before and make my "citation" list as that StackExchange Q&A indicate.

@ngoj1 That sounds promising! If a little slow. 

> I asked a librarian on the UBC library's online chat but unfortunately they weren't familiar with this, so if I can't figure out a better method anytime soon I will probably visit them in person since it's difficult to explain our question through text only.

I found out the hard way that the UBC online library chat service is just some global consortium of librarians and rarely a UBC librarian; I would suggest trying to track down someone at the library who is familiar with ISI and/or reference software if possible.

**Comment by ngoj1** [comment]
Email logs from today 16 July 2024 regarding the reference assistance form about citation downloading.

![Rachael_16July2024_1](https://github.com/user-attachments/assets/a907ffc7-4224-4b3d-9b3b-d9b465df363d)
![Rachael_16July2024_2](https://github.com/user-attachments/assets/4c4b25e9-42f2-4c6b-8268-84a00ec23fcd)
![Rachael_16July2024_3](https://github.com/user-attachments/assets/88ad908c-4c36-48ed-8a4b-2328835aface)

**Comment by lizzieinvancouver** [comment]
The email did not forward well so here are the missing parts:

![Screenshot 2024-07-16 at 4 32 50 PM](https://github.com/user-attachments/assets/58cdea0f-4da0-49df-ab7a-bdc698790b56)

![Screenshot 2024-07-16 at 4 33 09 PM](https://github.com/user-attachments/assets/06f85cc2-3aa5-40ad-94e6-3b9605bba3bb)

**Comment by ngoj1** [comment]
What we want isn't actually the list of citing articles, since that's the list of articles that cite the papers in our list of 700 or so (or 200 or so), but we want the list of references. I asked the librarians about this and Aubrey from Woodward says that no database she knows of has a function where you can extract references easily, BUT we can try **Citation Chaser** which only works with papers within the Lens.org purview but given its extensiveness I feel like it's worth a try and if a few papers get left behind we can go back and manually get those references.

Here are screenshots of what she said:
![Aubrey_23July2024_1](https://github.com/user-attachments/assets/b0b9e49d-8ae1-4f6d-abbe-fe09a7f46de3)
![Aubrey_23July2024_2](https://github.com/user-attachments/assets/13559b0a-cd92-4695-a8b3-0d4db0ee6755)
![Aubrey_23July2024_3](https://github.com/user-attachments/assets/2bab39d3-68fb-4183-a5d2-96c5bdbd0fd3)

**Comment by ngoj1** [comment]
After trying out citation chaser, there are only around 20 articles that don't get included when you transfer the DOIs into Citation chaser, either because 1) they don't have an associated DOI when exported from Web of Science, or 2) the journal doesn't fall under Lens.org. This means we get around 180 papers for the combined search of Janzen 1971 OR 1970 AND predator satiat* and around 730 papers for density dependen*.

However, when I search for the papers that are referenced within these lists of 180 or 730, only a small subsection of them can be exported because of the Lens.org affiliation. For example, from 730 papers that cite either Janzen 1970 or 1971 and are about density dependen*, there are roughly 32,000 references listed in total, but only 4000 of them are exportable from Citation chaser.

My current dilemma is trying to get these remaining 28,000 references that aren't included in this reference report from Citation chaser, but I'm not sure how that would be possible without resorting to the earlier method of exporting 50 papers from Web of Science at a time and doing everything through Web of Science. I may just send in a reference assistance form again, but we can talk about this tomorrow at the meeting!

**Comment by ngoj1** [comment]
After sending out a new reference assistance form about this issue of missing 28,000 of those papers due to Lens.org not supporting them:
![image](https://github.com/user-attachments/assets/6f14d327-ec65-4ca0-9fc0-8820a92d4483)

From Paul Lesack:
![image](https://github.com/user-attachments/assets/f484ae1e-0abf-4fbf-a60e-b25c4ebaf34a)

And from Brett Dimond:
![image](https://github.com/user-attachments/assets/37a1e6cb-a779-4747-8ff7-9514a1f0057d)

I'll be giving this GSscraper thing a try to see what it can do for us over the weekend most probably.

**Comment by ngoj1** [comment]
I took a look through GSscraper and I think from the start there are some problems. Here's the code you need to run it, which involves a lot of installation and updating. 
```
install.packages(c("shiny", "RCurl", "purrr", "dplyr", "shinyjs", "shinyWidgets", "textclean", "stringi", "magrittr", "zen4R", "shinyalert", "shinybusy", "strex"))
devtools::install_github("hrbrmstr/cfhttr")
library(stringr)
shiny::runGitHub("GSscraper", "nealhaddaway", subdir = "inst/shiny-examples/GSscraper")
```
It launches an app where you get to search for keywords or authors or etc. like this:
![image](https://github.com/user-attachments/assets/8b0dad0a-7ba8-4877-ab22-06a5c01345c4)

It basically looks like Web of Science search engine, but worse...and I don't think you can supply a list of DOIs into the shiny app and retrieve the sum of references from all of those DOIs. So I think we might just have to do as Paul Lesack says and book an appointment with Siobhan Schenk and ask for help on this.  😔

**Comment by lizzieinvancouver** [comment]
@ngoj1 I am sorry this has been such a huge pain! I think you have done the best possible and we have good documentation here for someone else to pick it up.

**Comment by lizzieinvancouver** [comment]
8 September 2024 email from Janneke:

I had a very useful meeting with the ETH bibliometrics specialists, where I outline what we wanted to do. They thought this was largely feasible, and are willing to help us do it. Here's a very brief summary of what I learned.


    They have access (and would use) a database called Dimensions.AI. They also use Web of Science, but apparently Dimensions provides more features and ETH pays to get everyone better access / more control (which I could technically wield, but I think they are more effective).
    To help us, they need from us:
        The key words used to define the two theories / bodies of literature.
            These should include all the booleans (e.g. and, not and or)
            Best if we test these ourselves (do we get the papers we expect) to minimize back and forths
            They will use these to search abstract, title (and maybe keywords), but not text (not easily possible).
        Also, what we consider the key citations that started the fields (and obviously which they represent). They can maybe do different sets of searches with these.
        They also want to know which fields of research both fields would be in (check dimensions website). See here - [Browse Fields of Research (ANZSRC 2020) for Publications - Dimensions](https://app.dimensions.ai/browse/categories/publication/for). 
        They also want to know, do we care what publication types? Options are: article, chapter, proceeding, preprint, monograph, edited book, seminar.
    They will then:
        Identify papers defining the two theories (from keywords)
        Quantify the cross citation (how often do papers in one group cite the other)
        Quantify the intra citation (how often do papers in one group cite papers in the same group)
        They can also send us a list of the papers that actually do cite both!!!
        I forgot to ask (but will ask) if this can be done by year, which I think is partly what makes it complicated (I think in each year it is only fair to ask if papers cite papers that were published earlier for these metrics)
    They may also try a cool visualization of within / between citation (this may be easier if lit is defined by the few key citations). I was thinking this could be cool for the foundational papers.
    Timing – they are currently busy, but if we send them in the next couple of weeks, could get on the project in October / November.

**Comment by lizzieinvancouver** [comment]
From Janneke yesterday: Here is a google doc (DELETED -- find in my notposting folder) where I'm collecting our key citations and key words to identify papers.

**Comment by lizzieinvancouver** [comment]
From discussions with Janneke today ... 

- Janzen and Connell papers are good for capturing the JC world
- May be harder to define Janzen as the masting paper so need to add keywords and/or adding papers (add Koenig? AND/OR Silvertown 1980)

**Comment by lizzieinvancouver** [comment]
Also, I found this [website](https://www.scottbot.net/HIAL/index.html@p=39432.html) which seems to show how to download the data we need (so not sure why this was such a struggle with the library last summer!). Basically: 

1. Go to ISI as described above and get the list you want.
2. Click on export and pick plain text.
3. Select the export function that lets you export the cited references (max of 500 at a time). Here's my screenshot:

<img width="1292" alt="Screenshot 2024-11-12 at 11 23 43 AM" src="https://github.com/user-attachments/assets/2022ff75-8125-4dc5-9b79-e60198d9cc7b">

Then we to pull them together (the website linked above has info on how to do it) and use GEPHI or something to visualize the results.
